python/private/pypi/generate_whl_library_build_bazel.bzl

```diff
@@ -14,406 +14,64 @@
-    }},
-    constraint_values = {constraint_values},
-    visibility = ["//visibility:private"],
-)""".format(
-                    name = p.replace("cp3", "python_3."),
-                    minor_version = abi[len("cp3"):],
-                    constraint_values = constraint_values_str,
-                ),
-            )
-        else:
-            additional_content.append(
-                """\
-config_setting(
-    name = "is_{name}",
-    constraint_values = {constraint_values},
-    visibility = ["//visibility:private"],
-)""".format(
-                    name = p.replace("cp3", "python_3."),
-                    constraint_values = constraint_values_str,
-                ),
-            )
-
-    return loads, "\n\n".join(additional_content)
+_RENDER = {
+    "copy_executables": render.dict,
+    "copy_files": render.dict,
+    "data": render.list,
+    "data_exclude": render.list,
+    "dependencies": render.list,
+    "dependencies_by_platform": lambda x: render.dict(x, value_repr = render.list),
+    "entry_points": render.dict,
+    "group_deps": render.list,
+    "srcs_exclude": render.list,
+    "tags": render.list,
+}
 
 def generate_whl_library_build_bazel(
         *,
-        dep_template,
-        whl_name,
-        dependencies,
-        dependencies_by_platform,
-        data_exclude,
-        tags,
-        entry_points,
         annotation = None,
-        group_name = None,
-        group_deps = []):
+        **kwargs):
     """Generate a BUILD file for an unzipped Wheel
 
     Args:
-        dep_template: the dependency template that should be used for dependency lists.
-        whl_name: the whl_name that this is generated for.
-        dependencies: a list of PyPI packages that are dependencies to the py_library.
-        dependencies_by_platform: a dict[str, list] of PyPI packages that may vary by platform.
-        data_exclude: more patterns to exclude from the data attribute of generated py_library rules.
-        tags: list of tags to apply to generated py_library rules.
-        entry_points: A dict of entry points to add py_binary rules for.
         annotation: The annotation for the build file.
-        group_name: Optional[str]; name of the dependency group (if any) which contains this library.
-          If set, this library will behave as a shim to group implementation rules which will provide
-          simultaneously installed dependencies which would otherwise form a cycle.
-        group_deps: List[str]; names of fellow members of the group (if any). These will be excluded
-          from generated deps lists so as to avoid direct cycles. These dependencies will be provided
-          at runtime by the group rules which wrap this library and its fellows together.
+        **kwargs: Extra args serialized to be passed to the
+            {obj}`whl_library_targets`.
 
     Returns:
         A complete BUILD file as a string
     """
 
+    kwargs["name"] = "unused"
```

Comment:
Instead of setting name=unused, how about setting it to the name of the package being processed?

Reply:
Decided to use the whl basename instead as the name, because it allows us to be more flexible in the future - like setup target_compatible_with based on the filename.